### PR TITLE
Documentation overhaul

### PR DESCRIPTION
**Pull request by ccoenen**
This Pull request aims to achieve

- a shorter, more welcoming README (along with removal of doctoc)
- all information in `/docs` (integrates a few things that were only on the wiki)
- consistency across documents
- extending a few things (default location CHANGELOG that helps people find where we actually document changes, for example).

### CODE REVIEW

File: docs/configuration-env-vars.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+| `CMD_LDAP_USERIDFIELD` | `uidNumber` or `uid` or `sAMAccountName` | The LDAP field which is used uniquely identify a user on CodiMD |
+| `CMD_LDAP_USERNAMEFIELD` | Fallback to userid | The LDAP field which is used as the username on CodiMD |
+| `CMD_LDAP_TLS_CA` | `server-cert.pem, root.pem` | Root CA for LDAP TLS in PEM format (use comma to separate) |
+| `CMD_LDAP_PROVIDERNAME` | `My institution` | Optional name to be displayed at login form indicating the LDAP provider |
+| `CMD_SAML_IDPSSOURL` | `https://idp.example.com/sso` | authentication endpoint of IdP. for details, see [guide](docs/guides/auth/saml-onelogin.md). |
+| `CMD_SAML_IDPCERT` | `/path/to/cert.pem` | certificate file path of IdP in PEM format |
+| `CMD_SAML_ISSUER` | no example | identity of the service provider (optional, default: serverurl)" |
+| `CMD_SAML_IDENTIFIERFORMAT` | no example | name identifier format (optional, default: `urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress`) |
+| `CMD_SAML_GROUPATTRIBUTE` | `memberOf` | attribute name for group list (optional) |
+| `CMD_SAML_REQUIREDGROUPS` | `Hackmd-users` | group names that allowed (use vertical bar to separate) (optional) |
```
Comment (SISheogorath) [inline docs/configuration-env-vars.md]: Here we still have a `Hackmd-users` appearance

File: docs/guides/auth/saml.md [hunk 2]
```diff
@@ -51,35 +51,35 @@ The basic procedure is the same as the case of OneLogin which is mentioned in [O
...
+    ```
   * environment variables
-    ````
-    HMD_SAML_GROUPATTRIBUTE=memberOf
-    HMD_SAML_REQUIREDGROUPS=hackmd-users|board-members
-    HMD_SAML_EXTERNALGROUPS=temporary-staff
-    ````
+    ```sh
+    CMD_SAML_GROUPATTRIBUTE=memberOf
+    CMD_SAML_REQUIREDGROUPS=hackmd-users|board-members
```
Comment (SISheogorath) [inline docs/guides/auth/saml.md]: This should be changed as the json example above

File: docs/slide-options.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
+Slide Separators
+===
+
+If you're getting started with reveal.js slides, there are a few things you need to know.
+
+There are two types of slides, those that transition horizontally and those that transition vertically (subslides).
+
+The following separators are used for each in the hackmd syntax:
```
Comment (SISheogorath) [inline docs/slide-options.md]: This should be `CodiMD syntax`

**Comment by ccoenen** [comment]
I fixed all the places you mentioned and a few more. I also broke up the configuration docs into bits, so they are easier to link to.

**Pull request merged**